.clang-tidy

```diff
@@ -0,0 +1,73 @@
+FormatStyle: file
+
+Checks: |
+  clang-diagnostic-*,
+  clang-analyzer-*,
+  boost-*,
+  bugprone-*,
+  -bugprone-easily-swappable-parameters,
+  clang-analyzer-*,
+  cppcoreguidelines-*,
+  -cppcoreguidelines-non-private-member-variables-in-classes,
+  -cppcoreguidelines-special-member-functions,
+  -cppcoreguidelines-avoid-magic-numbers,
+  -cppcoreguidelines-macro-usage,
+  -cppcoreguidelines-pro-type-reinterpret-cast,
+  -cppcoreguidelines-pro-bounds-constant-array-index,
+  google-*,
+  -google-readability-todo,
+  -google-build-using-namespace,
+  misc-*,
+  -misc-no-recursion,
+  -misc-non-private-member-variables-in-classes,
+  modernize-*,
+  -modernize-use-trailing-return-type,
+  performance-*,
+  -performance-no-int-to-ptr,
```

Comment:
See why this is excluded. However, reading https://releases.llvm.org/13.0.0/tools/clang/tools/extra/docs/clang-tidy/checks/performance-no-int-to-ptr.html, I was wondering whether carefully looking at the relevant places there is some room for improvement. As I interpret it from the link above, there might be ways to give the compiler more transparent information and potential for optimization.

Reply:
I did some research on this and would stay with the current implementation handling the tagged pointers. I think we would have to cast to `char*` anyway to change the LSB with pointer arithmetic and LLVM uses the same scheme as we are using in [Error.h](https://llvm.org/doxygen/Support_2Error_8h_source.html) (CTRL+F `void setChecked(bool V)`).

Reply:
Fully trust your judgement on this and agree that it seems reasonable to keep it that way. 👍🏻 feel free to resolve this.

Reply:
I resolve as discussed